core/src/main/java/com/datastax/oss/driver/internal/core/type/codec/MapCodec.java

```diff
@@ -134,18 +134,28 @@ public ByteBuffer encode(
       int size = input.getInt();
       Map<KeyT, ValueT> result = Maps.newLinkedHashMapWithExpectedSize(size);
       for (int i = 0; i < size; i++) {
+        KeyT key;
```

Comment:
If we are handling Map shouldn't we handle that for Tuple and Udt as well?

Reply:
`TupleCodec` and `UdtCodec` already handle null values.

UDT field names don't appear in the encoded representation. And there's no way they can ever be null anyway.